systemtest/src/test/java/io/strimzi/systemtest/upgrade/regular/StrimziUpgradeST.java

```diff
@@ -201,6 +166,32 @@ private void performUpgrade(BundleVersionModificationData upgradeData) throws IO
         verifyProcedure(upgradeData, testStorage.getContinuousProducerName(), testStorage.getContinuousConsumerName(), TestConstants.CO_NAMESPACE, wasUTOUsedBefore);
     }
 
+    private void performUpgradeAcrossVersions(TestStorage testStorage) throws IOException {
+        // Check if UTO is used before changing the CO -> used for check for KafkaTopics
+        boolean wasUTOUsedBefore = StUtils.isUnidirectionalTopicOperatorUsed(TestConstants.CO_NAMESPACE, eoSelector);
+
+        // Upgrade CO
+        changeClusterOperator(acrossUpgradeData, TestConstants.CO_NAMESPACE);
+
+        // Wait till first upgrade finished
+        controllerPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(TestConstants.CO_NAMESPACE, controllerSelector, 3, controllerPods);
+        brokerPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(TestConstants.CO_NAMESPACE, brokerSelector, 3, brokerPods);
+        eoPods = DeploymentUtils.waitTillDepHasRolled(TestConstants.CO_NAMESPACE, KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
```

Comment:
I expect that this will not work if the Kafka version is unsupported? You have to first change the Kafka version before waiting for a rolling update. Or how does it handle such a situation?

Reply:
Yeah I explained what is wrong with the test here: https://github.com/strimzi/strimzi-kafka-operator/issues/10495#issuecomment-2310817304

The parsing of Kafka version needs to be fixed, so the test will not be "flaky" and it will test exactly what it should.
In case that this check will be there, it will somehow work now, but when the real _unsupported_ Kafka version will be parsed there, it will fail the test.